src/rdkafka_op.c

```diff
@@ -482,25 +491,55 @@ rd_kafka_op_t *rd_kafka_op_new_cb (rd_kafka_t *rk,
 
 
 /**
- * @brief Reply to 'rko' re-using the same rko.
+ * @brief Reply to 'rko' re-using the same rko with rko_err
+ *        specified by \p err. rko_error is set to NULL.
+ *
  * If there is no replyq the rko is destroyed.
  *
  * @returns 1 if op was enqueued, else 0 and rko is destroyed.
  */
-int rd_kafka_op_reply (rd_kafka_op_t *rko, rd_kafka_resp_err_t err) {
+int rd_kafka_op_reply (rd_kafka_op_t *rko,
+                       rd_kafka_resp_err_t err) {
 
         if (!rko->rko_replyq.q) {
-		rd_kafka_op_destroy(rko);
+                rd_kafka_op_destroy(rko);
                 return 0;
-	}
+        }
 
-	rko->rko_type |= (rko->rko_op_cb ? RD_KAFKA_OP_CB : RD_KAFKA_OP_REPLY);
+        rko->rko_type |= (rko->rko_op_cb ? RD_KAFKA_OP_CB : RD_KAFKA_OP_REPLY);
         rko->rko_err   = err;
+        rko->rko_error = NULL;
 
 	return rd_kafka_replyq_enq(&rko->rko_replyq, rko, 0);
 }
 
 
+/**
+ * @brief Reply to 'rko' re-using the same rko with rko_error specified
+ *        by \p error (may be NULL) and rko_err set to the corresponding
+ *        error code.
+ *
+ * If there is no replyq the rko is destroyed.
+ *
+ * @returns 1 if op was enqueued, else 0 and rko is destroyed.
+ */
+int rd_kafka_op_error_reply (rd_kafka_op_t *rko,
+                             rd_kafka_error_t *error) {
+
+        if (!rko->rko_replyq.q) {
+                rd_kafka_op_destroy(rko);
```

Comment:
error is leaked here.

Reply:
it's not, `rd_afka_op_destroy` `RD_IF_FREE`'s the error.

Reply:
But it hasnt been assigned to rko_error yet.

Reply:
you're good at this!